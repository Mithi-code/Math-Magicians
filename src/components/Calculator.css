.calculator {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.result {
  background-color: rgb(189, 185, 216);
  width: 400px;
  height: 50px;
  display: flex;
  padding: 0 5px;
  align-items: center;
  justify-content: end;
  color: #fff;
  font-size: 20px;
}

table {
  width: 400px;
  border: 1px solid rgb(160, 157, 157);
}

tr {
  border: 1px solid rgb(160, 157, 157);
  height: 50px;
  background-color: #ebe8e8;
}

button {
  outline: none;
  border: none;
  background-color: unset;
  width: 100%;
  font-size: 20px;
}

td {
  height: 40px;
  width: 50px;
}

td:last-child {
  background-color: #ffa500;
}
