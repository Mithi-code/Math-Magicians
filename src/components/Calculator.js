import React, { Component } from 'react';
import './Calculator.css';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends Component {
  render() {
    return (
      <div className='calculator'>
        <div className='result'>0</div>
        <table>
          <tr>
            <td>AC</td>
            <td>+/-</td>
            <td>%</td>
            <td>+</td>
          </tr>
          <tr>
            <td>7</td>
            <td>8</td>
            <td>9</td>
            <td>Ã—</td>
          </tr>
          <tr>
            <td>4</td>
            <td>5</td>
            <td>6</td>
            <td>-</td>
          </tr>
          <tr>
            <td>1</td>
            <td>2</td>
            <td>3</td>
            <td>+</td>
          </tr>
          <tr>
            <td colSpan={2}>0</td>
            <td>.</td>
            <td>=</td>
          </tr>
        </table>
      </div>
    );
  }
}

export default Calculator;
