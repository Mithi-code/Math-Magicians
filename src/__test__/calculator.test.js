import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe ('Calculator Tests', ()=> {

it('Render works correctly', () => {
  const tree= renderer.create(Calculator).toJSON();
  expect(tree).toMatchSnapshot();
});

});