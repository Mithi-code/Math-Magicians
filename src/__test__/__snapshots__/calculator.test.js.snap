// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Tests Render works correctly 1`] = `null`;
